builtin/providers/google/provider.go

```diff
@@ -158,3 +158,31 @@ func getRegionFromZone(zone string) string {
 	}
 	return ""
 }
+
+// getRegion reads the "region" field from the given resource data and falls
+// back to the provider's value if not given. If the provider's value is not
+// given, an error is returned.
+func getRegion(d *schema.ResourceData, config *Config) (string, error) {
```

Comment:
Thanks for moving these in in here, `resource_compute_address.go` definitely wasn't the place for `getRegion`.